app/src/main/java/com/igordotsenko/dotsenkorssreader/ItemListActivity.java

```diff
@@ -19,290 +15,165 @@
         setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
-        currentChannelId = getIntent().getLongExtra(Channel.ID, -1);
-
-        //Broadcast reciever initialization
-        localBroadcastManager = LocalBroadcastManager.getInstance(this);
-        broadcastReceiver = new BroadcastReceiver() {
-            @Override
-            public void onReceive(Context context, Intent intent) {
-                //Show toast with message from RefreshService
-                if ( intent.getAction().equals(RefreshService.AUTOREFRESH_MESSAGE_ACTION + currentChannelId ) ) {
-                    handleAutoRefreshMessage(intent);
-                }
-                //Get filtered list of new items, add them to list, update recyclerview
-                if ( intent.getAction().equals(RefreshService.AUTOREFRESH_RESULT_ACTION + currentChannelId) ) {
-                    handleAutoRefreshResult(intent);
-                }
-            }
-        };
-
-        //Intent filter for interacting with RefreshService initialization
-        IntentFilter filter = new IntentFilter();
-        filter.addAction(RefreshService.AUTOREFRESH_RESULT_ACTION + currentChannelId);
-        filter.addAction(RefreshService.AUTOREFRESH_MESSAGE_ACTION + currentChannelId);
-        localBroadcastManager.registerReceiver(broadcastReceiver, filter);
-
-        //Start RefreshSercive
-        Intent intent = new Intent(ItemListActivity.this, RefreshService.class);
-        startService(intent.putExtra(Channel.ID, currentChannelId));
+        mCurrentChannelId = getIntent().getLongExtra(Channel.ID, -1);
 
         //SwipeRefreshLayout initialization
-        swipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.item_list_swiperefresh_layout);
-        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+        mSwipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.item_list_swiperefresh_layout);
+        mSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
             @Override
             public void onRefresh() {
-                Handler handler = new Handler(Looper.getMainLooper()) {
-                    @Override
-                    public void handleMessage(Message message) {
-                        super.handleMessage(message);
-                        Toast.makeText(getBaseContext(), message.getData().getString(RefreshingRunnable.MESSAGE_TAG), Toast.LENGTH_SHORT).show();
-                        updateItemList();
-                        swipeRefreshLayout.setRefreshing(false);
-                    }
-                };
-
-                Thread thread = new Thread(new RefreshingRunnable(handler.obtainMessage()));
-                thread.start();
+                ContentResolver.requestSync(
+                        MainActivity.sAccount,
+                        ReaderContentProvider.ContractClass.AUTHORITY,
+                        Bundle.EMPTY);
```

Comment:
With this bundle object SyncAdapter will not do force sync immediately. 
You need to put this flags with true values in extras:
ContentResolver.SYNC_EXTRAS_EXPEDITED
ContentResolver.SYNC_EXTRAS_MANUAL